arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -120,8 +120,7 @@ static inline u32_t _size_to_mpu_rasr_size(u32_t size)
 		return REGION_4G;
 	}
 
-	size = 1 << (32 - __builtin_clz(size - 1));
-	return (32 - __builtin_clz(size) - 2) << 1;
+	return (32 - __builtin_clz(size - 1) - 2 + 1) << 1;
```

Comment:
What about replacing `<< 1` by `<< MPU_RASR_SIZE_Pos`?

Reply:
@pizi-nordic I also added the relevant "masking".
And an explanatory comment in the function documentation about the rounding-up to next power-of-2.

Reply:
Great!